File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorAntipattern/Inbound/UdpSocket.java
package ServiceLocatorAntipattern.Inbound;

import ServiceLocatorAntipattern.ApplicationRoot;

import java.util.Random;

public class UdpSocket implements IInputSocket {
  public boolean receive(byte[] frameData) {
    ApplicationRoot.context.getComponent(Random.class).nextBytes(frameData); //stable dependency!
    return true;
  }
}

File: CSharp/DependencyInjection/DependencyInjection/ServiceLocatorDIAntipattern/_2_ServiceLocator/Inbound/UdpSocket.cs
using System;
using Unity;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Inbound
{
  internal interface IInputSocket
  {
    bool Receive(out byte[] frameData);
  }

  class UdpSocket : IInputSocket
  {
    public bool Receive(out byte[] frameData)
    {
      frameData = new byte[100];
      ApplicationRoot.Context.Resolve<Random>().NextBytes(frameData); //stable dependency!
      return true;
    }
  }
}